### PR TITLE
Issue 130 atmosphere perspective (photo gallery)

### PR DESCRIPTION
**Pull request by ismailsunni**
Issue: https://jira.camptocamp.com/browse/GSREN3D-130

- Add photo gallery component
- Add pinia store for its visibility and mock service for sample photos
- Hide and show the gallery if the button is toggled
 
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "story:preview": "histoire preview"
   },
   "dependencies": {
+    "@heroicons/vue": "^2.0.11",
```
Comment (fgravin) [inline package.json:30]: ok but will change I guess when the UI is ready

File: src/components/home/PhotoGallery.vue [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  photos: null as null | PhotoModel[],
+  galleryShown: Boolean,
+})
+
+const photoUrls = computed(() => {
+  return state.photos?.map((p) => p.url)
+})
+
+onMounted(async () => {
+  state.photos = await apiClientService.fetchPhotos()
```
[the remark this answers is not in the document]
Comment (fgravin) [reply src/components/home/PhotoGallery.vue:19]: not sure it's async though but maybe `imports` are async ? never tried actually :)

**Review by fgravin — APPROVED** [review]
Thanks for the work, it looks nice !
I think there will be some architecture work to optimize the loading of the assets but we can check that afterwards.
I recommend you to have a look on what Vite offers in that regard though.

File: src/components/ui/UiPhotoGalery.vue [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      class="flex flex-row justify-center items-center px-2 py-4 gap-4 w-11 h-7 bg-white rounded-t-xl"
+    >
+      <ChevronDownIcon :class="{ hidden: !galleryShown }"></ChevronDownIcon>
+      <ChevronUpIcon :class="{ hidden: galleryShown }"></ChevronUpIcon>
+    </button>
+    <div
+      class="flex flex-row items-start p-3 gap-3 w-auto h-56 bg-white rounded-l-xl rounded-r-xl rounded-t-none rounded-b-none"
+      :class="{ hidden: !galleryShown }"
+    >
+      <img
```
[the remark this answers is not in the document]
Comment (ismailsunni) [reply src/components/ui/UiPhotoGalery.vue:36]: I am trying to follow some approaches that I found:
- https://thewebdev.info/2022/03/12/how-to-preload-images-using-vue-js/ --> Adding rel="preload" in the image tag
- Adding 
  ```
   let image = new Image(); 
   image.src = 'something.png'.
   ```

Both don't work on my testing (do not give an obvious performance). Most of the time, the maps render slower than the images.

Anyway, I will keep this in mind and address this later when the time comes (CMIIW, early optimization is not a good practice). For now, I just reduce the photos' size.

Thanks for review, it opens a new window for me :)

**Pull request merged**